crates/python_utilities/src/ast_printer.rs

```diff
@@ -164,6 +175,148 @@ fn emit_body<'a>(body: &'a [Stmt], str_buffer: &mut WritingBuffer<'a>) {
     }
 }
 
+/**
+ * CustomDisplay represents a mutable pretty printing.
+ * WritingBuffer keep track of the indentation state, so we can't just return string.
+ * This allows, e.g. a nested list items to be double-indented.
+ *
+ * defer - When true, it does not push to str_buffer, and just returns the String value.
+ *         This functionality is partially implemented since we need it only to pick up
+ *         the function name.
+ */
+trait CustomDisplay {
+    fn custom_fmt(&self, str_buffer: &mut WritingBuffer, defer: bool) -> String;
+}
+
+impl CustomDisplay for ast::Expr {
+    fn custom_fmt(&self, str_buffer: &mut WritingBuffer, defer: bool) -> String {
+        fn push(str_buffer: &mut WritingBuffer, defer: bool, s: String) -> String {
+            if !defer {
+                str_buffer.push(Cow::Owned(s.clone()));
```

Comment:
can you make `fn cow_str(s: &str) -> Cow<String> { Cow::Owned(s.to_string()) }` and then just call this?

then you can do `cow_str(s.to_str())` here and below `cow_str(", ")` or `cow_str("]")` so it is easier to read.

Reply:
Addressed in 64150be9f93ffc954a02cfebda6f65460d294c17